api/modules/file/file.service.js

```diff
@@ -47,6 +47,9 @@ class FileService {
     if (csvRows.length > 100) {
       throw new Error('Uploaded file has more than 100 keywords');
     }
+    if (!csvRows.length) {
```

Comment:
Minor preference to check for 0 before checking if the value is more than 0 but less than 100:

```js
if (!csvRows.length) {
  //...
}

if (csvRows.length > 100) {
  //...
}
```

It looks more logical in that order, at least to me :-)